test/matrixAssignment.cpp

```diff
@@ -30,6 +30,7 @@ int main()
     for(int i=0; i<9; i++) {
         TEST(isnan(m_nan.data()[i]));
     }
+    TEST(m_nan.isAllNan());
```

Comment:
Also add a test for the negative result, to make sure it doesn't always return `true`